sql/moz-fx-data-shared-prod/mozilla_vpn_derived/subscription_events_live/view.sql

```diff
@@ -1,89 +1,106 @@
 CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.mozilla_vpn_derived.subscription_events_live`
 AS
-WITH json_arrays AS (
+WITH all_subscriptions AS (
   SELECT
     *,
     TO_JSON_STRING(promotion_codes) AS json_promotion_codes
   FROM
-    `moz-fx-data-shared-prod`.mozilla_vpn_derived.all_subscriptions_v1
+    mozdata.mozilla_vpn.all_subscriptions
+),
+previous_active_subscription_ids AS (
+  SELECT
+    active_date + 1 AS event_date,
+    subscription_id,
+  FROM
+    mozdata.mozilla_vpn.active_subscription_ids
+),
+current_active_subscription_ids AS (
+  SELECT
+    active_date AS event_date,
+    subscription_id,
+  FROM
+    mozdata.mozilla_vpn.active_subscription_ids
+),
+events AS (
+  SELECT
+    event_date,
+    subscription_id,
+    IF(previous_active_subscription_ids.subscription_id IS NULL, "New", "Cancelled") AS event_type,
+  FROM
+    current_active_subscription_ids
+  FULL JOIN
+    previous_active_subscription_ids
+  USING
+    (event_date, subscription_id)
+  WHERE
+    previous_active_subscription_ids.subscription_id IS NULL
+    OR current_active_subscription_ids IS NULL
 )
```

Comment:
I just realized my suggested alternate approach won't handle gaps in subscriptions.  Here's another variation that should properly handle gaps:
```suggestion
new_events AS (
  SELECT
    subscription_id,
    "New" AS event_type,
    active_date AS event_date,
  FROM
    mozdata.mozilla_vpn.active_subscription_ids
  QUALIFY
    LAG(active_date) OVER (PARTITION BY subscription_id ORDER BY active_date) IS DISTINCT FROM (active_date - 1)
),
cancelled_events AS (
  SELECT
    subscription_id,
    "Cancelled" AS event_type,
    active_date + 1 AS event_date,
  FROM
    mozdata.mozilla_vpn.active_subscription_ids
  QUALIFY
    LEAD(active_date) OVER (PARTITION BY subscription_id ORDER BY active_date) IS DISTINCT FROM (active_date + 1)
),
events AS (
  SELECT * FROM new_events
  UNION ALL
  SELECT * FROM cancelled_events
)
```

Reply:
instead of `LAG(active_date) OVER (PARTITION BY subscription_id ORDER BY active_date) IS DISTINCT FROM (active_date - 1)` how would you feel about `MIN(active_date) OVER (PARTITION BY subscription_id) = active_date`?